core/worker.cc

```diff
@@ -407,16 +410,31 @@ WorkerPauser::WorkerPauser() {
   if (is_any_worker_running()) {
     for (int wid = 0; wid < Worker::kMaxWorkers; wid++) {
       if (is_worker_running(wid)) {
-	workers_paused_.push_back(wid);
-	VLOG(1) << "*** Pausing Worker " << wid << " ***";
-	pause_worker(wid);
-	}
+        workers_paused_.push_back(wid);
+        VLOG(1) << "*** Pausing Worker " << wid << " ***";
+        pause_worker(wid);
+      }
     }
   }
 }
 
-WorkerPauser:: ~WorkerPauser() {
+WorkerPauser::~WorkerPauser() {
+  if (!workers_paused_.empty()) {
+    bess::run_global_resume_hooks(false);
+  }
+
```

Comment:
I noticed in my testing that there should actually be a call made to attach_orphans() made before workers are resumed as is done in when the ResumeAll command is processed. Otherwise, if there are no subsequent calls made to ResumeAll when no workers are running, orphans will remain unattached. So, you can either add the call to attach_orphans() here or I can provide a fix after this code merges.

Reply:
Great catch. I can fold the change into this PR

Reply:
Fixed